### PR TITLE
Conn1099

### PR DESCRIPTION
**Pull request by sailajaa**

### CODE REVIEW

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/LoginServiceImpl.java [hunk 1]
```diff
@@ -80,9 +80,11 @@ public LoginServiceImpl() {
     public UserLogin login(Login login) throws UserLoginException {
 
         UserLogin user = userLoginDAO.login(login);
+        if (user != null) {
         log.debug("db user name: ".concat(user.getUserName()));
```
Comment (msweaver) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/LoginServiceImpl.java]: formatting required.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/constant/NavigationConstant.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (msweaver) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/constant/NavigationConstant.java]: needs license.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java [hunk 3]
```diff
@@ -27,10 +27,11 @@
 package gov.hhs.fha.nhinc.admingui.managed;
 
 /*import gov.hhs.fha.nhinc.admingui.jee.jsf.UserAuthorizationListener;
-import gov.hhs.fha.nhinc.admingui.model.Login;
```
Comment (msweaver) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java:30]: remove commented out lines.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java [hunk 4]
```diff
@@ -57,153 +58,124 @@
...
+	@Autowired
+	private LoginService loginService;
+
+	/**
+	 * Gets the user name.
+	 * 
+	 * @return the user name
+	 */
+	public String getUserName() {
+		return userName;
```
Comment (msweaver) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java]: the formatting for this file looks wrong. It looks like tabs when it should only be a 4 spaces.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/dao/UserLoginDAOImpl.java [hunk 5]
```diff
@@ -62,8 +62,13 @@
     public UserLogin login(Login login) {
         Session session = this.sessionFactory.getCurrentSession();
         Query query = session.createQuery("from UserLogin where userName = :userName");
-        query.setParameter("userName", login.getUserName());
-        return (UserLogin) query.list().get(0);
+        query.setParameter("userName", login.getUserName());          
+		List list = query.list();
```
Comment (jsmith2) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/dao/UserLoginDAOImpl.java]: Why are we asking for a list here?  shouldn't it be query.uniqueResult()?

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/resources/Message.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+javax.faces.component.UIInput.REQUIRED=UserName and/or Password are Required.
```
Comment (msweaver) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/resources/Message.properties]: I don't see where this resource is used in the code. Is this a JSF level constant? Are we overwriting all JSF required messages to reference user name and password? That doesn't seem right.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java [hunk 7]
```diff
@@ -57,153 +58,124 @@
...
+
+	/**
+	 * Instantiates a new login bean.
+	 */
+	public LoginBean() {
+
+	}
+
+	/**
+	 * Invoke patient.
```
Comment (jsmith2) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java]: can we update this comment? I should have done this in a prior commit.  change "Invoke patient"  to something else.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/dao/UserLoginDAOImpl.java [hunk 8]
```diff
@@ -43,31 +45,31 @@
...
-     * @see gov.hhs.fha.nhinc.admingui.hibernate.dao.UserLoginDAO#login(gov.hhs.fha.nhinc.admingui.model.Login)
+     * @see gov.hhs.fha.nhinc.admingui.hibernate.dao.UserLoginDAO#login(gov.hhs.fha .nhinc.admingui.model.Login)
      */
     @Override
     public UserLogin login(Login login) {
         Session session = this.sessionFactory.getCurrentSession();
         Query query = session.createQuery("from UserLogin where userName = :userName");
         query.setParameter("userName", login.getUserName());
-        return (UserLogin) query.list().get(0);
+        return (UserLogin) query.uniqueResult();
```
Comment (jsmith2) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/dao/UserLoginDAOImpl.java]: Not sure why this wasn't caught before but we don't close the session here.  Need to surround with a try-catch-finally statement.

**Pull request merged**